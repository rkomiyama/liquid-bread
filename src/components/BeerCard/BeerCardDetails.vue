<template>
  <v-layout>
    <v-flex xs6>
      <h3>Ingredients</h3>
      <h4>Malt</h4>
      <v-layout v-for="malt in ingredients.malt" :key="malt.name">
        <v-flex xs6>{{ malt.name }}</v-flex>
        <v-flex xs6>{{ malt.amount.value }} {{ malt.amount.unit }}</v-flex>
      </v-layout>
      <h4>Hops</h4>
      <v-layout v-for="hops in ingredients.hops" :key="hops.name + hops.add">
        <v-flex xs6>{{ hops.name }} ({{ hops.add }})</v-flex>
        <v-flex xs6>{{ hops.amount.value }} {{ hops.amount.unit }}, {{ hops.attribute }}</v-flex>
      </v-layout>
      <h4>Yeast</h4>
      <div>{{ ingredients.yeast }}</div>
    </v-flex>
    <v-flex xs6>
      <h3>Method</h3>
      <h4>Fermentation</h4>
      <v-layout>
        <v-flex xs6>Temp</v-flex>
        <v-flex xs6>{{ method.fermentation.temp.value }} {{ method.fermentation.temp.unit }}</v-flex>
      </v-layout>
      <h4>Mash temp</h4>
      <div v-for="(mash_temp, i) in method.mash_temp" :key="i">
        <v-layout>
          <v-flex xs6>Duration</v-flex>
          <v-flex xs6>{{ mash_temp.duration }}</v-flex>
        </v-layout>
        <v-layout>
          <v-flex xs6>Temp</v-flex>
          <v-flex xs6>{{ mash_temp.temp.value }} {{ mash_temp.temp.unit }}</v-flex>
        </v-layout>
      </div>
      <h4>Twist</h4>
      <div>{{ method.twist ? method.twist : 'N/A' }}</div>
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  name: "BeerCardDetails",
  props: {
    ingredients: {
      type: Object,
      default: () => {}
    },
    method: {
      type: Object,
      default: () => {}
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
