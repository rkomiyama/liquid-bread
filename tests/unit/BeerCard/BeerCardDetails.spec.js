import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import BeerCardDetails from "@/components/BeerCard/BeerCardDetails.vue";
import beerData from "#/mockData/beerData.json";

Vue.use(Vuetify);

let wrapper;

describe("BeerCardDetails.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(BeerCardDetails, {
      propsData: { ingredients: beerData.ingredients, method: beerData.method }
    });
  });

  it("BeerCardDetails has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
