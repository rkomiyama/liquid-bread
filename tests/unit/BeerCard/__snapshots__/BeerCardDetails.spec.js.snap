// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCardDetails.vue BeerCardDetails has same HTML structure 1`] = `
<v-layout-stub tag="div">
  <v-flex-stub tag="div" xs6="">
    <h3>Ingredients</h3>
    <h4>Malt</h4>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Maris Otter Extra Pale</v-flex-stub>
      <v-flex-stub tag="div" xs6="">3.3 kilograms</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Caramalt</v-flex-stub>
      <v-flex-stub tag="div" xs6="">0.2 kilograms</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Munich</v-flex-stub>
      <v-flex-stub tag="div" xs6="">0.4 kilograms</v-flex-stub>
    </v-layout-stub>
    <h4>Hops</h4>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Fuggles (start)</v-flex-stub>
      <v-flex-stub tag="div" xs6="">25 grams, bitter</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">First Gold (start)</v-flex-stub>
      <v-flex-stub tag="div" xs6="">25 grams, bitter</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Fuggles (middle)</v-flex-stub>
      <v-flex-stub tag="div" xs6="">37.5 grams, flavour</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">First Gold (middle)</v-flex-stub>
      <v-flex-stub tag="div" xs6="">37.5 grams, flavour</v-flex-stub>
    </v-layout-stub>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Cascade (end)</v-flex-stub>
      <v-flex-stub tag="div" xs6="">37.5 grams, flavour</v-flex-stub>
    </v-layout-stub>
    <h4>Yeast</h4>
    <div>Wyeast 1056 - American Aleâ„¢</div>
  </v-flex-stub>
  <v-flex-stub tag="div" xs6="">
    <h3>Method</h3>
    <h4>Fermentation</h4>
    <v-layout-stub tag="div">
      <v-flex-stub tag="div" xs6="">Temp</v-flex-stub>
      <v-flex-stub tag="div" xs6="">19 celsius</v-flex-stub>
    </v-layout-stub>
    <h4>Mash temp</h4>
    <div>
      <v-layout-stub tag="div">
        <v-flex-stub tag="div" xs6="">Duration</v-flex-stub>
        <v-flex-stub tag="div" xs6="">75</v-flex-stub>
      </v-layout-stub>
      <v-layout-stub tag="div">
        <v-flex-stub tag="div" xs6="">Temp</v-flex-stub>
        <v-flex-stub tag="div" xs6="">64 celsius</v-flex-stub>
      </v-layout-stub>
    </div>
    <h4>Twist</h4>
    <div>N/A</div>
  </v-flex-stub>
</v-layout-stub>
`;
